code/modules/mob/living/say.dm

```diff
@@ -329,7 +329,12 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	return 1
 
 /mob/living/proc/can_speak_vocal(message) //Check AFTER handling of xeno and ling channels
-	if(HAS_TRAIT(src, TRAIT_MUTE) && get_message_language(message) != /datum/language/signlanguage)
+	var/obj/item/bodypart/leftarm = get_bodypart(BODY_ZONE_L_ARM)
+	var/obj/item/bodypart/rightarm = get_bodypart(BODY_ZONE_R_ARM)
+	if(HAS_TRAIT(src, TRAIT_MUTE) && get_selected_language() != /datum/language/signlanguage)
+		return 0
+	
+	if ((rightarm == null || rightarm.is_disabled()) && (leftarm == null ||leftarm.is_disabled()) && get_selected_language() == /datum/language/signlanguage)
 		return 0
```

Comment:
doing it this way will still runtime error because its still trying to run the function on rightarm and leftarm even if they're null. Something more like this might work better:
```suggestion
	if (get_selected_language() == /datum/language/signlanguage)
		var/left_disabled = FALSE
		var/right_disabled = FALSE
		if (istype(leftarm)) // Need to check if the arms exist first before checking if they are disabled or else it will runtime
			if (leftarm.is_disabled())
				left_disabled = TRUE
		else
			left_disabled = TRUE
		if (istype(rightarm))
			if (rightarm.is_disabled())
				right_disabled = TRUE
		else
			right_disabled = TRUE
		if (left_disabled && right_disabled) // We want this to only return false if both arms are either missing or disabled since you could technically sign one-handed.
			return 0
```
(Can someone else please look at this to see if it makes sense or if there's a cleaner way to do it.)

Reply:
in fact you could even move the assigning of the leftarm and rightarm underneath the first check if you did it like this. I _think_ it will work but its kind of ugly